plugins/inputs/jboss/jboss.go

```diff
@@ -0,0 +1,1222 @@
+	Subsystem     map[string]interface{} `json:"subsystem"`
+}
+
+// WebMetrics  Web Modules related metrics
+type WebMetrics struct {
+	ActiveSessions    string                 `json:"active-sessions"`
+	ContextRoot       string                 `json:"context-root"`
+	ExpiredSessions   string                 `json:"expired-sessions"`
+	MaxActiveSessions string                 `json:"max-active-sessions"`
+	SessionsCreated   string                 `json:"sessions-created"`
+	Servlet           map[string]interface{} `json:"servlet"`
+}
+
+// JBoss the main collectod struct
+type JBoss struct {
+	Servers []string
+	Metrics []string
+
+	Username string
+	Password string
+
+	ExecAsDomain bool `toml:"exec_as_domain"`
+
+	Authorization string
+
+	ResponseTimeout internal.Duration
+
+	// Path to CA file
+	SSLCA string `toml:"ssl_ca"`
+	// Path to host cert file
+	SSLCert string `toml:"ssl_cert"`
+	// Path to cert key file
+	SSLKey string `toml:"ssl_key"`
+	// Use SSL but skip chain & host verification
+	InsecureSkipVerify bool
+
+	client HTTPClient
+}
+
+// HTTPClient HTTP client struct
+type HTTPClient interface {
+	// Returns the result of an http request
+	//
+	// Parameters:
+	// req: HTTP request object
+	//
+	// Returns:
+	// http.Response:  HTTP respons object
+	// error        :  Any error that may have occurred
+	MakeRequest(req *http.Request) (*http.Response, error)
+
+	SetHTTPClient(client *http.Client)
+	HTTPClient() *http.Client
+}
+
+// RealHTTPClient the HTTP client handler
+type RealHTTPClient struct {
+	client *http.Client
+}
+
+// MakeRequest do an HTTP request
+func (c *RealHTTPClient) MakeRequest(req *http.Request) (*http.Response, error) {
+	return c.client.Do(req)
+}
+
+// SetHTTPClient set http client
+func (c *RealHTTPClient) SetHTTPClient(client *http.Client) {
+	c.client = client
+}
+
+// HTTPClient return the HTTP client
+func (c *RealHTTPClient) HTTPClient() *http.Client {
+	return c.client
+}
+
+var sampleConfig = `
+  # Config for get statistics from JBoss AS
+  servers = [
+    "http://[jboss-server-ip]:9090/management",
+  ]
+	## Execution Mode
+	exec_as_domain = false
+  ## Username and password
+  username = ""
+  password = ""
+	## authorization mode could be "basic" or "digest"
+  authorization = "digest"
+
+  ## Optional SSL Config
+  # ssl_ca = "/etc/telegraf/ca.pem"
+  # ssl_cert = "/etc/telegraf/cert.pem"
+  # ssl_key = "/etc/telegraf/key.pem"
+  ## Use SSL but skip chain & host verification
+  # insecure_skip_verify = false
+	## Metric selection
+	metrics =[
+		"jvm",
+		"web_con",
+		"deployment",
+		"database",
+		"jms",
+	]
+`
+
+// SampleConfig returns a sample configuration block
+func (*JBoss) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description just returns a short description of the JBoss plugin
+func (*JBoss) Description() string {
+	return "Telegraf plugin for gathering metrics from JBoss AS"
+}
+
+func (h *JBoss) checkAuth(host string, uri string) error {
+	url := h.Servers[0]
```

Comment:
Always uses the first server?

Reply:
This method belongs to the legacy code from previous contributor, and I'm right now just reviewing it. seems no longer used ... still reviewing